### PR TITLE
Make dotty use scala-infra for CI.

### PR DESCRIPTION
**Pull request by DarkDimius**
@adriaanm Thanks for huge help in getting this working :+1:

### CODE REVIEW

File: scripts/jobs/validate/scalastyle [hunk 1]
```diff
@@ -0,0 +1,6 @@
+baseDir=${WORKSPACE-`pwd`}
```
Comment (retronym) [inline scripts/jobs/validate/scalastyle]: I think these scripts will need `#!/bin/bash -e` to ensure the job fails.

File: scripts/jobs/validate/junit [hunk 2]
```diff
@@ -0,0 +1,6 @@
+baseDir=${WORKSPACE-`pwd`}
+scriptsDir="$baseDir/scripts"
+. $scriptsDir/common
+
+update scala scala
+sbt -Ddotty.travis.build=yes update compile test
```
Comment (retronym) [inline scripts/jobs/validate/junit]: `-no-colors` will avoid ugly ANSI codes in the output.

Comment (retronym) [inline scripts/jobs/validate/junit]: Oh, I've just seen https://github.com/scala/scala-jenkins-infra/pull/57.

**Comment by DarkDimius** [comment]
All the builds will fail until commits from this branch get in.
@odersky you need to update your setup to be in sync with b671424, unfortunatelly we cannot use `cd ..` in CI anymore, so scala/scala went to a subfolder.

**Comment by DarkDimius** [comment]
@odersky please review and merge it to unblock other PRs

**Comment by retronym** [comment]
@DarkDimius Not sure if you say my comment above, but I don't think you are correctly propagating the exit codes of the `sbt` process in your scripts.

> I think these scripts will need #!/bin/bash -e to ensure the job fails.

**Comment by DarkDimius** [comment]
@retronym I can add `-e`, but can you explain why current script is incorrect?
While integrating this PR there were moments when `partest` failed, and failure was propagated.

**Comment by adriaanm** [comment]
The `-e` option matters when you have multiple steps. Currently, only the last one will influence the exit code, so intermediate ones could fail and you won't be notified.

**Comment by DarkDimius** [comment]
@adriaanm thanks for explaining!

**Pull request merged**